src/main/java/de/tum/in/www1/artemis/service/ExamService.java

```diff
@@ -297,4 +302,24 @@ private Exercise selectRandomExercise(SecureRandom random, ExerciseGroup exercis
         int randomIndex = random.nextInt(exercises.size());
         return exercises.get(randomIndex);
     }
+
+    /**
+     * Starts all the exercises of all the student exams of an exam
+     *
+     * @param examId exam to which the student exams belong
+     * @return list of generated participations
+     */
+    @Transactional
```

Comment:
Please do not use Transactional here and rather use a query such as 

`studentExamRepository.findWithExercisesAndParticipationsByExamId`

Then make sure that only the participations for the user of the student exam is loaded, otherwise your "isEmpty()" check does not make any sense later on

Reply:
I'll add comments and we have to improve this later on, but as others need this functionality for testing, we'll merge this now